modules/c++/sys/source/OSUnix.cpp

```diff
@@ -189,9 +189,14 @@ std::string sys::OSUnix::getTempName(const std::string& path,
     else
     {
         name = tempname;
+        sys::File newFile(name, sys::File::READ_AND_WRITE, sys::File::CREATE);
         free(tempname);
```

Comment:
You'll get a warning about an unused variable this way.  If you just leave it unnamed...

```
sys::File (name, ...);
```

You'll still get the side effect you want (constructor will create the file) without the warning.